packages/snaps-jest/src/environment.ts

```diff
@@ -0,0 +1,161 @@
+import { EnvironmentContext, JestEnvironmentConfig } from '@jest/environment';
+import { assert, createModuleLogger } from '@metamask/utils';
+import { Server } from 'http';
+import NodeEnvironment from 'jest-environment-node';
+import { AddressInfo } from 'net';
+import { Browser } from 'puppeteer';
+import { remote } from 'webdriverio';
+
+import { rootLogger, startServer } from './internals';
+import { getOptions, SnapsEnvironmentOptions } from './options';
+
+/* eslint-disable */
+declare global {
+  const browser: WebdriverIO.Browser;
+  const snapsEnvironment: SnapsEnvironment;
+}
+/* eslint-enable */
+
+export class SnapsEnvironment extends NodeEnvironment {
+  // `browser` is always set in the environment setup function. To avoid needing
+  // to check for `undefined` everywhere, we use a type assertion here.
+  browser!: WebdriverIO.Browser;
+
+  #options: SnapsEnvironmentOptions;
+
+  #server: Server | undefined;
+
+  /**
+   * Constructor.
+   *
+   * @param options - The environment options.
+   * @param context - The environment context.
+   */
+  constructor(options: JestEnvironmentConfig, context: EnvironmentContext) {
+    super(options, context);
+    this.#options = getOptions(options.projectConfig.testEnvironmentOptions);
+  }
+
+  /**
+   * Set up the environment. This starts the built-in HTTP server, and creates a
+   * new browser instance.
+   */
+  async setup() {
+    await super.setup();
+
+    if (this.#options.server.enabled) {
+      this.#server = await startServer(this.#options.server);
+    }
+
+    const args = [];
+    if (this.#options.browser.headless) {
+      args.push('--headless');
+    }
+
+    this.browser = await remote({
+      logLevel: 'error',
+      capabilities: {
+        browserName: 'chrome',
```

Comment:
Are we only supporting Chrome for now?

Ideally we would be able to E2E test in FF too

Reply:
Maybe something for a future PR? It will add a lot of complexity to run in two browsers.